flow/project.py

```diff
@@ -1133,9 +1109,17 @@ def _generate_id(self, aggregate, operation_name=None):
         else:
             op_string = operation_name
 
+        # Ids are generated based on the project root directory. However, while
+        # testing, we must use a mocked value of the root directory. This is a
+        # somewhat hacky solution.
+        root_directory = (
+            project.root_directory()
+            if not getattr(project, "_mock", False)
+            else project._mock_root_directory()
+        )
```

Comment:
I proposed an alternative to this in `generate_template_reference_data.py`, would be nice to get rid of if possible but depends on whether my proposal would work.

Reply:
The proposal won't work - see other comment. I will resolve this and continue the conversation on the other thread.